nevergrad/benchmark/experiments.py

```diff
@@ -1373,33 +1373,69 @@ def double_o_seven(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
     Dimension 6 and 7.
     Budget 100 to 3200
     """
     seedg = create_seed_generator(seed)
-    optims: tp.List[tp.Any] = ["NaiveTBPSA", "PSO", "DE", "LhsDE", "RandomSearch", "NGO", "Shiwa", "DiagonalCMA",
-                               "CMA", "OnePlusOne", "TwoPointsDE", "NGOpt10"]
+    optims = get_optimizers("structure", "structured_moo", seed=next(seedg))
     optims += [ng.families.DifferentialEvolution(multiobjective_adaptation=False).set_name("DE-noadapt"),
-               ng.families.DifferentialEvolution(crossover="twopoints", multiobjective_adaptation=False).set_name(
-                   "TwoPointsDE-noadapt")]
+    ng.families.DifferentialEvolution(crossover="twopoints", multiobjective_adaptation=False).set_name(
+            "TwoPointsDE-noadapt")]
+    optims += ["DiscreteOnePlusOne", "DiscreteLenglerOnePlusOne"]
+    popsizes = [20, 40, 80]
+    optims += [ng.families.EvolutionStrategy(recombination_ratio=recomb, only_offsprings=only, popsize=pop,
+        offsprings=pop * 5)
+    for only in [True, False] for recomb in [0.1, .5] for pop in popsizes]
     mofuncs: tp.List[fbase.MultiExperiment] = []
+    dim = 2000 if hd else 7
     for name1, name2 in itertools.product(["sphere"], ["sphere", "hm"]):
-        mofuncs.append(fbase.MultiExperiment([ArtificialFunction(name1, block_dimension=7),
-                                              ArtificialFunction(name2, block_dimension=7)],
-                                             upper_bounds=[100, 100]))
-        mofuncs.append(fbase.MultiExperiment([ArtificialFunction(name1, block_dimension=6),
-                                              ArtificialFunction("sphere", block_dimension=6),
-                                              ArtificialFunction(name2, block_dimension=6)],
-                                             upper_bounds=[100, 100, 100.]))
+        mofuncs.append(fbase.MultiExperiment([ArtificialFunction(name1, block_dimension=dim),
+                    ArtificialFunction(name2, block_dimension=dim)] + (
+                        [ArtificialFunction(name1, block_dimension=dim),
+                        ArtificialFunction(name2,
+                            block_dimension=dim)] if
+                        many else []),
+                    upper_bounds=[100, 100] * (2 if many else 1)))
+        mofuncs.append(fbase.MultiExperiment([ArtificialFunction(name1, block_dimension=dim-1),
+                    ArtificialFunction("sphere", block_dimension=dim-1),
+                    ArtificialFunction(name2, block_dimension=dim-1)] +
+                    ([ArtificialFunction(name1, block_dimension=dim-1),
+                     ArtificialFunction("sphere",
+                         block_dimension=dim-1),
+                     ArtificialFunction(name2,
+                         block_dimension=dim-1)]
+                     if
+                     many
+                     else
+                     []),
```

Comment:
thats becoming extremely hard to read, can you rewrite in a more reader friendly way? :s

Reply:
Done :-)